lib/hunter/profile_cli.rb

```diff
@@ -39,13 +39,11 @@ def apply(node, identity)
           identity
         ]
         cmd = new(*flight_profile, *args)
-        cmd.run.tap do |result|
-          if result.success?
-            return result.stdout
-          else
-            puts "ERROR"
-          end
+
+        pid = Process.fork do
+          cmd.run
         end
+        Process.detach(pid)
```

Comment:
This introduces some problems.

- We are no longer tracking the exit status of the subprocess. We don't know if the process succeeded or not.
- If the process succeeds, we aren't returning the stdout of the process.
- If the process fails, we aren't letting anybody know that the process has failed. Granted, the old way of just printing `ERROR` wasn't very helpful, but getting rid of it is even less helpful.

As these are only things that will occur in the `apply` command fork, I don't think they're truly important to deal with right now. The command will be logging to `logger`, and as it's a command meant to be used for automation, we don't necessarily need to know whether it's succeeded or not: the user can check the logfile for that.

Also, while this _does_ solve the problem, I think that the solution _might_ be better solved in changing how Flight Profile starts its `apply` processes. Did you look into that at all? (No worries if not, but if you haven't then I'll put it in my notes to have a look at some other time)

Reply:
Most of why I was comfortable removing the tracking of subprocess success is that we don't use it for anything anyway, and returning the stdout of `apply` is largely useless. As it stands, you can still check the success of the command, but it's up to you to do that from wherever you call it from (since the output of `cmd.run` is what the function now returns).

>Also, while this does solve the problem, I think that the solution might be better solved in changing how Flight Profile starts its apply processes. Did you look into that at all?

I did but wasn't able to stop `flight-subprocess` waiting for it. It seems designed to use all of the timeout to wait for any zombie grandchild processes - https://github.com/openflighthpc/flight-subprocess/blob/main/lib/flight/subprocess/local.rb#L193 - and I wasn't able to come up with a fix on Hunter's side that would tell `flight-subprocess` that it doesn't need to do this. It seems that closing stdout for the `apply` process would do it, but that messes up `apply` and causes it to fail every time. It seems to keep running until stdout is closed, which only happens when the `apply` process fully completes.